libraries/AP_HAL_Linux/Scheduler.cpp

```diff
@@ -63,6 +63,13 @@ void Scheduler::init_realtime()
     // we don't run Replay in real-time...
     return;
 #endif
+#if APM_BUILD_TYPE(APM_BUILD_UNKNOWN)
+    // we opportunistically run examples/tools in realtime
```

Comment:
I still have the same opinion that I had previously: this is broken - the **tools** should declare if they need realtime or not, not an ifdef into the scheduler.

I don't mind if you add a parameter that's passed to init and it defaults to "no realtime" so you only need to change the vehicle code.

I also don't see how you plan to run examples on the server... only the most basic ones will be able to run.